### ISSUE TITLE
How to use other datasets?

### ISSUE THREAD

**Issue by ZhiyuanDang** [JunjieHu/koil#1]
Thanks for your great work!

The code is clear and specific.

I want to use other datasets in LIBSVM datasets. However, the dataset folder maybe misses some m files to construct the  ``dataset-name``_IdxAsso and ``dataset-name``_IdxCv.  

Could you please release the related code or some scripts?

**Comment by ZhiyuanDang** [comment]
I use crossvalind function in Matlab generate the dataset-name_IdxCv file.

But I have another problem: the results on l1 and l2 loss are totally different in the same hyperparameters. 

For example. in shuttle dataset, I set C is 2^7 and \gamma is 2^10 which states in your TNNLS paper, the results of l1 loss are same as the paper (RS++: .950, FIFO++: .956), but the l2 results are different, (RS++: .781, FIFO++: .778). It is weird.

Is there anything wrong with my settings?  Thanks!

**Comment by JunjieHu** [comment]
Hi @ZhiyuanDang 

It has been a long time since I wrote the code! 

I think the difference is mainly due to the [hyperparameters](https://github.com/JunjieHu/koil/blob/master/matlab/KOIL_kSize.m#L16), such as sigma in the RBF kernel, learning rate and cost penalty. It would be better to find the best sigma separately for the l2 loss and l1 loss [by grid-search in the cross-validation step](https://github.com/JunjieHu/koil/blob/master/matlab/CV_nKOIL.m#L50). 

In your experiment, did you search the sigma by using the l1 loss in the cross-validation step, and stick to that sigma for both losses in the training step? 

PS: If you want to search the hyperparameters, it's faster to run the C++ implementation. You can refer to the second example [here](https://github.com/JunjieHu/koil#c-implementation) by setting the loss type to l2 .

**Comment by ZhiyuanDang** [comment]
Yes, I use the C++ version of the code. I adopt the hyperparameters displayed in your TNNLS paper (Table 1), the results of l1 are the same as the ones provide in your paper (Table III), but the results of l2  mostly lower than the ones you provide in the paper.

In your experiment settings, the hyperparameter of l1 and l2 are the same or not? Thanks for your reply.

**Comment by JunjieHu** [comment]
I've lost access to the machine that I used to run those experiments, so I don't have the logs at hand to check. It has been a while and honestly I don't remember whether the same parameters are used for both losses, but I would recommend searching the best hyper-parameters C and gamma again separately for both losses on your processed data.

**Comment by ZhiyuanDang** [comment]
OK. Thanks for your suggestion.
